custom-event-duration/action.yml

```diff
@@ -0,0 +1,28 @@
+name: Send Duration Custom Event to Okay
+on:
+  workflow_call:
+    inputs:
+      event:
+        description: 'Name of the event'
+        required: true
+      event_unique_id:
+        description: 'The unique id for this event'
+        required: true
+      user:
+        description: 'The github login of the person'
+        required: false
+      metric_name:
+        description: 'The name of the duration metric'
+        required: true
+      metric_duration_ms:
+        description: 'How long it took (in milliseconds)'
+        required: true
+jobs:
+  okay-send-custom-event-duration-webhook:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Send Okay Webhook
+      env:
+        OKAY_API_TOKEN: ${{ secrets.OKAY_API_TOKEN }}
+      run: |
+        curl -X POST -H "Authorization: Bearer $OKAY_API_TOKEN" -H 'Content-Type: application/json' --data "{\"version\": \"1\", \"type\": \"duration\", \"event\": \"custom\",  \"uniqueKey\": [\"event_unique_id\"], \"customEventName\": \"${{ inputs.event }}\", \"timestamp\": \"$(date +%Y-%m-%dT%H:%M:%S%z)\", \"identity\": {\"user\": \"${{ inputs.user }}\", \"type\": \"sourceControlLogin\"}, \"properties\": {\"event_unique_id\": \"${{ inputs.event_unique_id }}\", \"${{ inputs.metric_name }}\": \"${{ inputs.metric_duration_ms }}\"}}" https://app.okayhq.com/api/events/v1
```

Comment:
for `duration` a top-level `durationMs` property is required.  though, now that i think about it, we might not need types due to the flexibility of the current schema.

my original thought was something like this:

```
{
  'event': 'custom',
  'customEventName': 'buildEvent',
  'type': 'duration',
  'durationMs': 100,
  'timestamp': '2022-03-01 00:00:00',
  'properties': {
     repo: 'app',
     org: 'foobar',
     build: 'tests'
  }
}
```

But we could just as easily do what you did and put the duration metric in properties.  In this case, we do not need the explicit `durationMs` prop or type.  wdyt?

Reply:
yeah, i think it'll be better to be consistent between count/duration/batch so not have durationMs as a top level field (i think it'll limit the flexibility long term).  now that i'm thinking about this more we might need some addition context in batch to be able to distinguish between counts and durations too

Reply:
ok, so how about this:

Type: duration (generates `buildEvent` duration metric and custom filters for `repo`, `org` and `build`)

```
{
  'event': 'custom',
  'customEventName': 'buildEvent',
  'type': 'duration',
  'timestamp': '2022-03-01 00:00:00',
  'properties': {
     durationMs: 100,                <----  required
     repo: 'app',
     org: 'foobar',
     build: 'tests'
  }
}
```

Type: count (generates `buildEvent` count metric and custom filters for `repo`, `org` and `build`)

```
{
  'event': 'custom',
  'customEventName': 'buildEvent',
  'type': 'count',
  'timestamp': '2022-03-01 00:00:00',
  'properties': {
     repo: 'app',
     org: 'foobar',
     build: 'tests'
  }
}
```

Type: batch (generates metrics for `flakeCount`, `totalCount`, `incidentMinutes` and custom filters for `repo`, `org` and `build`)

```
{
  'event': 'custom',
  'customEventName': 'buildStats',
  'type': 'batch',
  'timestamp': '2022-03-01 00:00:00',
  'properties': {
     uniqueKey: ['repo', 'org', 'build],
     metrics: ['flakeCount', 'totalCount', 'incidentMinutes'],
     repo: 'app',
     org: 'foobar',
     build: 'tests',
     flakeCount: 3,
     totalCount: 10,
     incidentMinutes: 35
  }
}
```

Reply:
i also think we'll ultimately provide the ability to create schemas for custom events, which can be used to generate additional metrics (e.g. `flakeCount / totalCount`)